project/AddLogTimestamps.scala

```diff
@@ -0,0 +1,64 @@
+package akka
+
+import java.io.PrintWriter
+import java.text.SimpleDateFormat
+
+import sbt._
+import Keys._
+import sbt.Def
+import sbt.internal.LogManager
+import sbt.internal.util.ConsoleOut
+
+object AddLogTimestamps extends AutoPlugin {
+  val enableTimestamps: Boolean = CliOption("akka.log.timestamps", false).get
+
+  override def requires: Plugins = plugins.JvmPlugin
+  override def trigger: PluginTrigger = allRequirements
+
+  override def projectSettings: Seq[Def.Setting[_]] = {
+    logManager := {
+      val original = logManager.value
+
+      if (enableTimestamps) {
+        val myOut = new PrintWriter(System.out) {
+          val dateTimeFormat = new SimpleDateFormat("MM-dd HH:mm:ss.SSS")
```

Comment:
could this be called concurrently? perhaps use `DateTimeFormatter`, since `SimpleDateFormat` is not thread safe